File: pset6/caesar.py
#!/usr/bin/env python3
import cs50
import sys

def main():
    cl_check()
    k = sys.argv[1]
    key = int(k)                    #convert from string to integer
    print("plaintext: ", end="")
    message = cs50.get_string()
    print("ciphertext: ", end="")
    cipherByCaesar(message, key)
        
def cl_check():
    if len(sys.argv) == 2 and int(sys.argv[1]) > 0:
        return
    else:
        sys.exit("Usage: ./caesar k (where k is a non-negative integer key for Caesar cipher)")
        

def cipherByCaesar(message, key):
    for char in message:
        if char.isalpha():
            code = ord(char)  # getting the integer equivalent of char
            if char.islower():
                code = (code - ord("a") + key) % 26 + ord("a")
            if char.isupper():
                code = (code - ord("A") + key) % 26 + ord("A")
            encrypted = chr(code)  # transform to the char from integer
        else:
            encrypted = char
        print("{}".format(encrypted), end="")
    print()
    
if __name__ == "__main__":
    main()

File: problem-set-2-/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>


void cipherbyCaesar(char* , int );

int main(int argc, string argv[]) {
    int key = 0;
    if ((argc != 2) || atoi(argv[1]) < 0 ) { // user would enter a command-line argument, if > 1(exception for program name) arguments or nothing = error
        printf ("Usage: ./caesar k (where k is a non-negative integer key for Caesar cipher)\n");
        return 1;
    }
    else {
         key = atoi(argv[1]);                                         // converts string to integer
    }
        printf ("plaintext: ");
        string expression = get_string();                             // get string from user
        printf("ciphertext: ");
        cipherbyCaesar(expression, key);    
    return 0;
}


void cipherbyCaesar (char* symb, int k) {
    for (int i = 0, length = strlen(symb); i < length; i++) {  
                                                                    // 26 is the number of letters in english alphabet
        if (isalpha(symb[i]) && isupper(symb[i]))                   //cycle for encryption uppercase letters
            printf("%c", (symb[i] - 'A' + k) % 26 + 'A');
            
        if (isalpha(symb[i]) && islower(symb[i]))                   //cycle for encryption lowecase letters
            printf("%c", (symb[i] - 'a' + k) % 26 + 'a');
            
        if (!isalpha(symb[i]))                                      // cycle for missing non-alphabetical characters
            printf("%c", symb[i]);
  
    }
        printf("\n");
    
}